### DISCUSSION TITLE
Getting 'Unexpected token' error when trying reuse an inactive assignment

### DISCUSSION THREAD

**Discussion by nashnj13** [community/community#120145] [GitHub Education]
### Hi everyone,

![image](https://github.com/community/community/assets/102609656/e7302ebe-9074-43b3-86a4-7138c981ba7a)

All I have done is select the drop-down boxes and click continue.

**Comment by samus-aran** [comment]
Hi @nashnj13 👋 

Are you still experiencing this?
If you are able to provide more details that would be much appreciated. What assignment are you trying to reuse? Are you reusing it by editing a single assignment or from the /classrooms page where you can reuse multiple assignments? What browser and version are you using?

**Reply by nashnj13** [reply]
I was not able to replicate this, so it may just have been a cache issue. However, I am still unable to edit or create teams for a reused assignment. Clicking reuse for multiple or just one assignment ends up with the same result. An assignment marked as inactive and when the edit button or 'create teams' button is clicked, sends me to a 500 error page.

I am using Chrome Version 124.0.6367.119 (Official Build) (64-bit)

**Comment by bharlowRHC** [comment]
@samus-aran I can provide some additional detail as I am also experiencing this and have multiple new assignments that are experiencing this error as seen above. Many of my older assignments that I have not made any modifications to I am able to copy. 

I have tried and failed to complete the Reuse Assignment feature due to this error in the following browsers on two different operating systems:

1. **Microsoft Edge (Mac):** Version 125.0.2535.92 (Official build) (arm64) - This is my primary browser
2. **Apple Safari (Mac):** Version 17.5 (19618.2.12.11.6) - This had never been logged into GitHub Classroom
3. **Microsoft Edge (Windows):** Version 125.0.2535.92 (Official build) (64-bit) - This use to be my primary computer but had to be relogged into Classroom.
4. **Google Chrome (Windows):** Version 126.0.6478.57 (Official build) (64-bit) - This had never been logged into GitHub Classroom.

The example assignment in the screenshots below is in an active classroom with students completing as of yesterday. I just created the assignment new yesterday because I could not copy the original due to this error. I am using the single Reuse Assignment button rather than the bulk reuse. This example, like the additional assignments from the staging classroom I use for setup/reuse of new classes, cannot be copied. They all result in a 404-error code as seen in the browser dev tools and nothing appears in the target classroom.

I have other new assignments that cannot be copied also, I just tested and took the screenshots using the same one. Interestingly, I can copy assignments that have not been modified in the UI since just after the rollout of the AutoGrading and Protect File features.

In terms of settings, example below is an active individual Python assignment created yesterday in GitHub Classroom where I am using the new Fork method in an organization created several weeks ago for the class. I do not have an IDE selected nor do students have admin permission. I do have the Feedback pull request selected. I have a combination of IO and Run Command Auto Grading tests set to manually run with several files flagged as protected.

Below are a set of screenshots that show what I am seeing, which shows the error the same as the original screenshot above, but with more detail. Let me know if you have questions as I can consistently recreate this error at this point using new assignments.

<img width="1968" alt="classroom-reuse-1" src="https://github.com/community/community/assets/37458144/c52838e4-9b6d-4c7a-9abd-938466444276">

<img width="1968" alt="classroom-reuse-2" src="https://github.com/community/community/assets/37458144/54df2032-6314-4820-ada9-0e87d32fc2d9">

<img width="1968" alt="classroom-reuse-3" src="https://github.com/community/community/assets/37458144/8edb58d2-cb96-4c5d-b384-ceb149caf787">

<img width="1324" alt="classroom-reuse-4" src="https://github.com/community/community/assets/37458144/7584de1d-4068-4fee-82d0-1829e9d9253f">

**Reply by bharlowRHC** [reply]
I created a new assignment today for a different exercise with a new repo and Classroom Exercise. This time I systematically tested the Reuse Assignment process. To see what worked and what did not work.

**Here is what was successful when using the Reuse Assignment:**

- If I did not have any test cases, I could copy the assignment successfully.
- If I had one Input/Output test case, I could copy the assignment successfully.
- If I had one Python grader test case, I could copy the assignment successfully. _I generally do not use this since I have different point values for different tests cases that cannot be accounted by this grader._

**Here is what resulted in error for the Reuse Assignment:**

- If I had two or more Input/Output test cases, it failed to copy with the previously reference error message.
- If I had one of the Run Command Auto Grading tests, it failed to copy with the previously referenced error message.

I tried toggling the "Every time a student submits" as well as the "Manual" to different settings for how they are run and that did not affect it. Below are the settings of the new assignment that caused the error state when using the Reuse Assignment feature. These are setup identical to a previous assignment that worked in the Spring Semester.

<img width="1375" alt="error-io1" src="https://github.com/community/community/assets/37458144/697eee80-c507-47b5-9b7f-baf92e582700">

<img width="1398" alt="error-io2" src="https://github.com/community/community/assets/37458144/6ab3ab29-1c5a-4dd3-9db3-028cb2357a5e">

<img width="1390" alt="error-ut1" src="https://github.com/community/community/assets/37458144/a7e351c9-d9b8-4d86-8b34-6c4274b482d8">

**Reply by samus-aran** [reply]
Hi @bharlowRHC 👋 

Thank you for this detailed report. We've shared this with the internal team to further investigate 👍

**Reply by nimonian** [reply]
@samus-aran Any update on this? I am, unable to reuse _any_ of my assignments.

**Comment by regisin** [comment]
Still an issue. I'm using Java with gradle tests. Everything works except reusing the assignment. I get a:

```JSON.parse: unexpected character at line 1 column 1 of the JSON data```

Error message.

**Comment by dolan-peter** [comment]
I'm having the same issue with my group assignments from a previous class too.
